Update provider to allow configuring resource group

wkirby opened the pull request:
This PR addresses #44 and #106 by allowing you to configure `resourceGroup` and 
`deploymentName` in `serverless.yml`.

```
provider:
  name: azure
  resourceGroup: your-resource-group
  deploymentName: your-deployment-name
```

We have been using this fork in production for several weeks with no issues.

jeevan-shikaram commented:
This conflicts with the `serverless remove` command. The remove command with can be found in `./remove/azureRemove.js` deletes the resource group.

I assume the original creators recognized that if they allowed a user to assign their own resource group, there will be a danger of the resource group being deleted.

So when a user specifies a resource group, the whole resource group shouldn't be deleted as there might be other resources inside it.

Inside `./remove/lib/deleteResourceGroup.js`:
`module.exports = {
  deleteResourceGroup () {
    return this.provider.Login()
      .then(() => this.provider.DeleteDeployment())
      .then(() => this.provider.DeleteResourceGroup());
  }
};`

This should be changed to detect if the resource group user-specified and should not delete the resource group.

wkirby commented:
I don’t see how this conflicts, as the `remove` command still works, and this change doesn’t affect default behavior.

I take your point about `remove` being a dangerous command, but I don’t see how it is any more dangerous than the current implementation, as there’s nothing stopping me from adding to the resource group created by serverless now.

I also don’t see that this PR is a good fit for addressing the concerns raised in #48 as the feature request to configure a custom resource group name is separate from the issue raised there.

It seems to me that this should be identified in the documentation, that using a resource group that includes services not created or managed by serverless has additional risk.

ihachani commented:
I think the problem raised in #48  can be solved by specifying the function name and keeping track of it.

nohemirojas commented:
Any progress on this issue? It's still open, I have the same problem with the resource group

loket commented:
Same here. This is my only blocker for using Serverless Framework as I need to follow strict naming conventions when deploying to Azure.

I see the point of using the `remove` command, but that's not an issue for me.

heina-chang commented:
Hello,
We are working on deploying some stuff to azure and would love to be able to control the resource group.  When do you plan to merge this in?  many thanks

tbarlow12 commented:
LGTM :shipit:
